### PR TITLE
Allowed pre-population of add specific query fields (main)

### PR DESCRIPTION
**Pull request by ganning127**
**With both `add` and `alias`**

Screenshot 2023-06-28 at 11 11 52 AM

**With only add**

Screenshot 2023-06-28 at 11 12 19 AM

**With only alias**

Screenshot 2023-06-28 at 11 12 42 AM

- Should we change `add` to `sqlStr`? Or is it fine the way it currently is?
- The URL populates with ?add=...&alias=.... as you type in the SQL string and alias on the edit page---happy to do a live demo of this

### CODE REVIEW

**Comment by korydraughn** [comment]
Is this actually supported by the REST API?

**Comment by ganning127** [comment]
> Is this actually supported by the REST API?

Yep, it's allowed. /admin endpoint allows us to add specific queries. This PR only pre populates the adding field, but requires the user themselves to click save, allowing them to change the alias name if needed.

**Comment by korydraughn** [comment]
You know what, seeing the URL in the picture threw me off.

Carry on.

**Comment by ganning127** [comment]
Just pushed a commit that changes the way the confirmation box looks to be more intuitive (can revert if needed)

**Saving new query**
Screenshot 2023-06-28 at 1 50 14 PM

**Deleting specific query**
Screenshot 2023-06-28 at 1 50 22 PM

File: src/views/SpecificQuery.js [hunk 1]
```diff
@@ -33,10 +33,12 @@ const useStyles = makeStyles((theme) => ({
 
 export const SpecificQuery = () => {
     if (!localStorage.getItem('zmt-token')) navigate('/');
+    const location = useLocation()
+    const params = new URLSearchParams(location.search)
     const classes = useStyles()
     const [status, setStatus] = useState('none')
-    const [newAlias, setNewAlias] = useState()
-    const [newSqlStr, setNewSqlStr] = useState()
+    const [newAlias, setNewAlias] = useState(params.get('alias') ? decodeURIComponent(params.get('alias')) : "");
```
Comment (trel) [inline src/views/SpecificQuery.js:40]: semicolon...

File: src/views/SpecificQuery.js [hunk 2]
```diff
@@ -156,8 +176,20 @@ export const SpecificQuery = () => {
                     <DialogContentText>{status} specific query {status === 'Adding' ? newAlias : aliasToDelete}?</DialogContentText>
```
Comment (trel) [inline src/views/SpecificQuery.js:176]: please add single quotes or square brackets around the displayed alias.

File: src/views/SpecificQuery.js (hunk 1 above)
Comment (ganning127) [reply src/views/SpecificQuery.js:40]: added semicolons to everything in that section. seems like the other pages/tabs (like `Server.js`) all have semicolons...

File: src/views/SpecificQuery.js (hunk 2 above)
Comment (ganning127) [reply src/views/SpecificQuery.js]: Added single quotes wherever we're displaying an alias on a pop-up box

**Comment by trel** [comment]
looking very good.  please squash.

**Comment by ganning127** [comment]
squashed to 1 commit

**Comment by trel** [comment]
commit message... consider...

`[169] allow pre-population when adding specific-queries`

**Comment by ganning127** [comment]
yep, i like that better. commit message has been updated to that

**Comment by trel** [comment]
cool - let's # it.

**Comment by ganning127** [comment]
added!

**Pull request merged**